tests/system/test_system.py

```diff
@@ -6,17 +6,281 @@
+        )
+        # connect to the test database
+        conn = Connection(Config.INSTANCE, self._db)
+        cursor = conn.cursor()
+
+        # execute several DML statements within one transaction
+        cursor.execute(
+            """
+INSERT INTO contacts (contact_id, first_name, last_name, email)
+VALUES (1, 'first-name', 'last-name', 'test.email@domen.ru')
+        """
+        )
+        cursor.execute(
+            """
+UPDATE contacts
+SET first_name = 'updated-first-name'
+WHERE first_name = 'first-name'
+"""
+        )
+        cursor.execute(
+            """
+UPDATE contacts
+SET email = 'test.email_updated@domen.ru'
+WHERE email = 'test.email@domen.ru'
+"""
+        )
+        conn.commit()
+
+        # read the resulting data from the database
+        cursor.execute("SELECT * FROM contacts")
+        got_rows = cursor.fetchall()
+        conn.commit()
+
+        self.assertEqual(got_rows, [want_row])
+
+    def test_rollback(self):
+        """Test rollbacking a transaction with several statements."""
+        want_row = (2, "first-name", "last-name", "test.email@domen.ru")
+        # connect to the test database
+        conn = Connection(Config.INSTANCE, self._db)
+        cursor = conn.cursor()
+
+        cursor.execute(
+            """
+INSERT INTO contacts (contact_id, first_name, last_name, email)
+VALUES (2, 'first-name', 'last-name', 'test.email@domen.ru')
+        """
+        )
+        conn.commit()
+
+        # execute several DMLs with one transaction
+        cursor.execute(
+            """
+UPDATE contacts
+SET first_name = 'updated-first-name'
+WHERE first_name = 'first-name'
+"""
+        )
+        cursor.execute(
+            """
+UPDATE contacts
+SET email = 'test.email_updated@domen.ru'
+WHERE email = 'test.email@domen.ru'
+"""
+        )
+        conn.rollback()
+
+        # read the resulting data from the database
+        cursor.execute("SELECT * FROM contacts")
+        got_rows = cursor.fetchall()
+        conn.commit()
+
+        self.assertEqual(got_rows, [want_row])
+
+    def test_autocommit_mode_change(self):
+        """Test auto committing a transaction on `autocommit` mode change."""
+        want_row = (
+            2,
+            "updated-first-name",
+            "last-name",
+            "test.email@domen.ru",
+        )
+        # connect to the test database
+        conn = Connection(Config.INSTANCE, self._db)
+        cursor = conn.cursor()
+
+        cursor.execute(
+            """
+INSERT INTO contacts (contact_id, first_name, last_name, email)
+VALUES (2, 'first-name', 'last-name', 'test.email@domen.ru')
+        """
+        )
+        cursor.execute(
+            """
+UPDATE contacts
+SET first_name = 'updated-first-name'
+WHERE first_name = 'first-name'
+"""
+        )
+        conn.autocommit = True
+
+        # read the resulting data from the database
+        cursor.execute("SELECT * FROM contacts")
+        got_rows = cursor.fetchall()
+        conn.commit()
```

Comment:
This should not be necessary (and now probably causes a warning), as the connection is in autocommit mode.